Offroad-iOS/Offroad-iOS/Presentation/CourseQuest/ViewControllers/CourseQuestViewController.swift

```diff
@@ -0,0 +1,194 @@
+//
+//  CourseQuestViewController.swift
+//  Offroad-iOS
+//
+//  Created by  정지원 on 5/13/25.
+//
+import UIKit
+
+import SnapKit
+import Then
+
+class CourseQuestViewController: UIViewController, UICollectionViewDelegate, UIGestureRecognizerDelegate {
+    
+    private let courseQuestView = CourseQuestView()
+    private let courseQuestDetailService = CourseQuestDetailService()
+    private var quests: [CourseQuestDetailPlaceDTO] = []
+    var questId: Int?
+    var deadline: String?
+    
+    private let collectionView = UICollectionView(frame: .zero, collectionViewLayout: UICollectionViewFlowLayout()).then {
+        let layout = UICollectionViewFlowLayout()
+        layout.itemSize = CGSize(width: UIScreen.main.bounds.width - 48, height: 97)
+        layout.minimumLineSpacing = 14
+        $0.collectionViewLayout = layout
+        $0.backgroundColor = .clear
+        $0.contentInset = .init(top: 16, left: 0, bottom: 16, right: 0)
+        $0.register(CourseQuestPlaceCell.self, forCellWithReuseIdentifier: "CourseQuestPlaceCell")
+    }
+    
+    override func loadView() {
+        view = courseQuestView
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        courseQuestView.courseQuestPlaceCollectionView.dataSource = self
+        courseQuestView.listContainerView.delegate = self
+        setupControlsTarget()
+        
+        if let deadline = deadline {
+            let ddayString = Self.dday(from: deadline)
+            let dateString = formattedDate(from: deadline)
+            courseQuestView.deadlineDateLabel.text = "퀘스트 마감일: \(dateString)"
+            courseQuestView.ddayBadgeLabel.text = ddayString
+        }
+        
+        if let questId = questId {
+            fetchCourseQuestDetail(questId: questId)
+        }
+    }
+    
+    private func fetchCourseQuestDetail(questId: Int) {
+        courseQuestDetailService.getCourseQuestDetail(questId: questId) { [weak self] result in
+            switch result {
+            case .success(let dto):
+                DispatchQueue.main.async {
```

Comment:
P4
클로저 안에서 `self`를 참조하는 건
`DispatchQueue.main.async{ ... }` 블럭 안에서만인 것 같은데, 
`[weak self]` 캡처리스트를 `DispatchQueue.main.async {` 코드 뒤로 옮기는 게 더 의도를 명확히 할 수 있을 것 같습니다!

```suggestion
        courseQuestDetailService.getCourseQuestDetail(questId: questId) { result in
            switch result {
            case .success(let dto):
                DispatchQueue.main.async { [weak self] in 
```